library/bignum_core.h

```diff
@@ -0,0 +1,157 @@
+
+/** Import X from unsigned binary data, big endian.
+ *
+ * The MPI needs to have enough limbs to store the full value (including any
+ * most significant zero bytes in the input).
+ *
+ * \param X      The address of the MPI.
+ *               May only be #NULL if \nx is 0 and \p buflen is 0.
+ * \param nx     The number of limbs of \p X.
+ * \param buf    The input buffer to import from.
+ *               May only be #NULL if \p buflen is 0.
+ * \param buflen The length in bytes of \p buf.
+ *
+ * \return       \c 0 if successful.
+ * \return       #MBEDTLS_ERR_MPI_BUFFER_TOO_SMALL if \p X isn't
+ *               large enough to hold the value in \p buf.
+ */
+int mbedtls_mpi_core_read_be( mbedtls_mpi_uint *X,
+                              size_t nx,
+                              const unsigned char *buf,
+                              size_t buflen );
+
+/** Export X into unsigned binary data, little endian.
+ *
+ * \note If \p buf is shorter than \p X the export is still successful if the
+ *       value held in \p X fits in the buffer (that is, if enough of the most
+ *       significant bytes of \p X are 0).
+ *
+ * \param X      The address of the MPI.
+ * \param nx     The number of limbs of \p X.
+ * \param buf    The output buffer to export to.
+ * \param buflen The length in bytes of \p buf.
+ *
+ * \return       \c 0 if successful.
+ * \return       #MBEDTLS_ERR_MPI_BUFFER_TOO_SMALL if \p buf isn't
+ *               large enough to hold the value of \p X.
+ */
+int mbedtls_mpi_core_write_le( const mbedtls_mpi_uint *X,
+                               size_t nx,
+                               unsigned char *buf,
+                               size_t buflen );
+
+/** Export X into unsigned binary data, big endian.
+ *
+ * \note If \p buf is shorter than \p X the export is still successful if the
+ *       value held in \p X fits in the buffer (that is, if enough of the most
+ *       significant bytes of \p X are 0).
+ *
+ * \param X      The address of the MPI.
+ * \param nx     The number of limbs of \p X.
+ * \param buf    The output buffer to export to.
+ * \param buflen The length in bytes of \p buf.
+ *
+ * \return       \c 0 if successful.
+ * \return       #MBEDTLS_ERR_MPI_BUFFER_TOO_SMALL if \p buf isn't
+ *               large enough to hold the value of \p X.
+ */
+int mbedtls_mpi_core_write_be( const mbedtls_mpi_uint *X,
+                               size_t nx,
+                               unsigned char *buf,
+                               size_t buflen );
+
+#define ciL    ( sizeof( mbedtls_mpi_uint ) )   /* chars in limb  */
+#define biL    ( ciL << 3 )                   /* bits  in limb  */
+#define biH    ( ciL << 2 )                   /* half limb size */
+
+/*
+ * Convert between bits/chars and number of limbs
+ * Divide first in order to avoid potential overflows
+ */
+#define BITS_TO_LIMBS(i)  ( ( i ) / biL + ( ( i ) % biL != 0 ) )
+#define CHARS_TO_LIMBS(i) ( ( i ) / ciL + ( ( i ) % ciL != 0 ) )
```

Comment:
I don't think these changes are correct, since [the coding standard](https://mbed-tls.readthedocs.io/en/latest/kb/development/mbedtls-coding-standards/) specifically gives the following example, with no spaces around the macro parameter when used in the expansion

```
#define GET_UINT32_LE( n, b, i )                        \
{                                                       \
    (n) = ( (uint32_t) (b)[(i)    ]       )             \
        | ( (uint32_t) (b)[(i) + 1] <<  8 )             \
        | ( (uint32_t) (b)[(i) + 2] << 16 )             \
        | ( (uint32_t) (b)[(i) + 3] << 24 );            \
}
```

So I think this should be

```suggestion
#define BITS_TO_LIMBS( i )  ( (i) / biL + ( (i) % biL != 0 ) )
#define CHARS_TO_LIMBS( i ) ( (i) / ciL + ( (i) % ciL != 0 ) )
```